src/debugger/Debugger.cc

```diff
@@ -375,14 +375,24 @@ void Debugger::Cmd::disasm(std::span<const TclObject> tokens, TclObject& result,
 
 void Debugger::Cmd::disasmBlob(std::span<const TclObject> tokens, TclObject& result) const
 {
-	checkNumArgs(tokens, 4, Prefix{2}, "value addr");
+	checkNumArgs(tokens, Between{4, 5}, Prefix{2}, "value addr ?symbols?");
 	std::span<const uint8_t> bin = tokens[2].getBinary();
 	auto len = instructionLength(bin);
 	if (!len || *len > bin.size()) {
 		throw CommandException("Blob does not contain a complete Z80 instruction");
 	}
 	std::string dasmOutput;
-	dasm(bin.subspan(0, *len), word(tokens[3].getInt(getInterpreter())), dasmOutput);
+	unsigned addr = tokens[3].getInt(getInterpreter());
+	dasm(bin.subspan(0, *len), word(addr), dasmOutput,
+		[&](std::string& output, uint16_t a) {
+			TclObject key(a);
+			auto label = tokens[4].getDictValue(getInterpreter(), key).getString();
+			if (!label.empty()) {
+				strAppend(output, label);
+			} else {
+				appendAddrAsHex(output, a);
+			}
+		});
 	dasmOutput.resize(19, ' ');
```

Comment:
Now you require to pass the full dictionary, in other words you _eagerly_ need to account for all possible labels, What if we allow to _lazily_ translate values to names. So instead of passing an (optional) dictionary to the `disasm_blob` function we pass the name of a proc that can translate a value to a string-representation. For example:
```
proc formatLabel {value} {
    if {exists $::sym($value)} {
        return $::sym($value)
    } else {
        return [format "#%04x" $value]
    }
}
```
This example uses the global `$::sym` array, this contains all symbols known by the openMSX symbol manager. But of course you can also create a proc that does a lookup in your own dict.